source/Octopus.Client.Tests/Operations/RegisterMachineOperationFixture.cs

```diff
@@ -205,5 +234,24 @@ public void ShouldNotUpdateExistingMachine()
                 && m.EnvironmentIds.First() == "environments-2"
                 && m.MachinePolicyId == "MachinePolicies-2")).ConfigureAwait(false);
         }
+
+        [Test]
+        public void ShouldThrowIfTenantsAndModeDisagree()
```

Comment:
Probably deserves a test for the alternate scenario where `TenantedDeploymentMode.Untenanted` but `Tenants` are supplied.